modules/cudaarithm/test/test_core.cpp

```diff
@@ -281,6 +281,8 @@ CUDA_TEST_P(Flip, Accuracy)
 
 CUDA_TEST_P(Flip, AccuracyInplace)
 {
+    size.width  = (size.width  >> 1) << 1; // in-place version only accepts even number
+    size.height = (size.height >> 1) << 1; // in-place version only accepts even number
```

Comment:
I propose to split tests for in-place and general cases and instantiate in-place cases with even size. It's more obvious behavior.

Reply:
@asmorkalov , do you mean create `DIFFERENT_EVEN_SIZES` list [here](https://github.com/opencv/opencv/blob/b39cd06249213220e802bb64260727711d9fc98c/modules/ts/include/opencv2/ts/cuda_test.hpp#L212) and instantiate inplace tests separately? I think it would make this test unnecessary complex.

Reply:
That's something debatable.  I first had the same feeling with @asmorkalov , since the log shows `113x113` which is not correct.
Then I saw the [the line](https://github.com/opencv/opencv/blob/b39cd06249213220e802bb64260727711d9fc98c/modules/ts/include/opencv2/ts/cuda_test.hpp#L212) that @mshabunin pointed, and thought the same thing. "It's only a single test which uses `DIFFERENT_EVEN_SIZES` "
Merging the idea will lead me to add single define in the `test_core.cpp` rather in `cuda_test.hpp`, but I need someone as an arbitor.

Reply:
Sorry, but I again changed my mind.
I thought it would be simple, but it's really making a another copy of flip test code just for the in-place version.
Now, there are one `Flip` test, but asmorkalov suggestion is to create `Flip` and `Flip_inplace` test case, which I don't prefer.
Can I keep the current implementation as-is, since it'll make things much easier ?

Reply:
Another idea is to keep sizes intact but add bad args handling to the test:
```.cpp
if (/* size is not even */)
{
    EXPECT_THROW(...);
    return;
}
// current test body
```